src/solvers/refinement/string_refinement.cpp

```diff
@@ -1124,13 +1127,17 @@ static void substitute_array_access(exprt &expr)
         "above"));
     array_exprt &array_expr=to_array_expr(index_expr.array());
 
-    // Empty arrays do not need to be substituted.
+    const typet &char_type = index_expr.array().type().subtype();
+
+    // Access to an empty array is undefined (non determinstic result)
     if(array_expr.operands().empty())
+    {
+      expr = symbol_generator("out_of_bound_access", char_type);
```

Comment:
Is the type important here? Why is it `char_type`?

Reply:
yes, this is an access in an array of characters so we are expecting the result to be of the type of the elements of the array